### DISCUSSION TITLE
Highlight matching pairs

### DISCUSSION THREAD

**Discussion by musjj** [helix-editor/helix#8489] [Ideas]
Right now helix only highlights the other pair:

![image](https://github.com/helix-editor/helix/assets/72612857/d6f7e32d-4e9e-45b2-b4fe-800e2631ea50)

It would be nice if *both* parts of the pair gets highlighted.

**Comment by CptPotato** [comment]
One issue with this is when using a block cursor you would not be able to see the cursor anymore if both parts use the highlighted style. This could only work with the bar cursor style.

**Reply by musjj** [reply]
Maybe the cursor could have a higher priority over the quote highlight. In neovim and emacs, pair highlights works fine even with a block cursor hovering over it.

**Reply by arunoruto** [reply]
AFAIK it is currently like this and it drives me nuts somehow...
![image](https://github.com/user-attachments/assets/6594025f-ec9d-4f22-8733-9a61364ef6ed)
I am still wondering how to highlight them differently
